src/Engine/ProtoCore/CodeGen.cs

```diff
@@ -2363,7 +2363,7 @@ protected bool InsideFunction()
 
         // used to manully emit "return = null" instruction if a function or language block does not have a return statement
         // there is update code involved in associativen code gen, so it is not implemented here
-        protected abstract void EmitReturnNull();
```

Comment:
this looks like an API break, it will break clients who implemented this method in their derived classes.

Reply:
Separated the optional parameter into with and without parameter to preserve the public API.